### PR TITLE
[Console] Add return type to OutputFormatterInterface::format()

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | N/A
| License       | MIT
| Doc PR        | N/A

Spotted while working on #42381. If we look at how `OutputFormatterInterface::format()` is used in the codebase, we can see that an implementation of that method is supposed to return something. Yet the interface does not declare a return value and the `NullOutputFormatter` implementation even has a `void` return type which does not make sense at all, imho.

This PR attempts to fix that.

### CODE REVIEW

**Comment by Tobion** [comment]
Could you also fix the test i mentioned in https://github.com/symfony/symfony/pull/42381#issuecomment-893069202 to make more sense please.

**Comment by derrabus** [comment]
Done. I'm really scratching my head over this class. It clearly does not behave as the author of the test thought it would. 🤔

**Review by Tobion — APPROVED** [review]

File: src/Symfony/Component/Console/Formatter/OutputFormatterInterface.php
Comment (wouterj) [inline src/Symfony/Component/Console/Formatter/OutputFormatterInterface.php:57]: I think this must be `@return string` and the null formatter should be `return $message;` instead.

https://github.com/symfony/symfony/blob/49a076a120147727ed3759c678c06d79ed922521/src/Symfony/Component/Console/Output/Output.php#L157

**Review by wouterj — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Console/Formatter/OutputFormatterInterface.php
Comment (derrabus) [reply src/Symfony/Component/Console/Formatter/OutputFormatterInterface.php:57]: Keep in mind that this is not an internal class. You are suggesting to significantly change its behavior here while I'm merely documenting the status quo.

Comment (wouterj) [reply src/Symfony/Component/Console/Formatter/OutputFormatterInterface.php:57]: Hmm, indeed. I'm not sure what is the best way forward.

If we officially document this contract as returning `string|null`, we have to update all code using this function to take care of the `null` return. Honestly, returning null (void actually) in `NullOutputFormatter` seems like a mistake and bug to me.

In fact, if code is build to use the normal `OutputFormatter`, it would be broken if `NullOutputFormatter` is used in the test. I can't think of any reasonable approach to build code to only work with `NullOutputFormatter`.

Comment (derrabus) [reply src/Symfony/Component/Console/Formatter/OutputFormatterInterface.php:57]: I understand. But `string|null` is what currently is returned from the implementations. Once this PR is merged, we can think about how we could narrow it down to just `string` in 6.0. By the way, I'd also like to do this for the `$message` parameter.

Comment (ro0NL) [reply src/Symfony/Component/Console/Formatter/OutputFormatterInterface.php:57]: Should we depreciate null in 5.4 then? Then we can keep the implied contract, which is string IMHO

Comment (fabpot) [reply src/Symfony/Component/Console/Formatter/OutputFormatterInterface.php:57]: I agree that the real issue is the implementation of `NullOutputFormatter` which was written before void was introduced. Conceptually, doing nothing was the same as returning `null` back then. Let's merge this one as is and see if we can deprecate `null` in a followup PR (same for `*message`).

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @derrabus.

**Pull request merged**
